### ISSUE TITLE
Always prefix id params with resource

### ISSUE THREAD

**Issue by paulcsmith** [luckyframework/lucky#659]
TL;DR `/users/:user_id` instead of `/users/:id` When generating routes with the `route` macro

I've been thinking about what the default path param should be for non-nest resources. In Rails/Phoenix/other frameworks it is typically just `id`. But when nested (for example `/users/:user_id/projects/:id` one path param gets prefixed with the resource name, and the other does not. I think this can lead to confusion which is why I like the idea in this issue: luckyframework/lucky#497

**I think we could go a step further though and always prefix the param with the resource**, e.g. `/users/:user_id` instead of `/users/:id`. The advantage of this is that it is always super clear what kind of id you are working with, but also it means that you can use a module that works in nested and non-nested actions. For example: https://gist.github.com/paulcsmith/101fc224fd59c5d3d8c9b70bef3a38ee

```crystal
module ProjectFinder
  def project
    # We would be able to use this method in nested and non-nested resources since 
    # the resource will be :project_id in either case
    ProjectQuery.new.user_id(current_user.id).find(project_id)
  end
end
```

It would also make it easier to change a controller from nested to non-nested and the param name would stay the same. Making it simpler to do.

**Comment by jwoertink** [comment]
Is this only referring to when you use `route`? Like, if I do:

```crystal
class User::Show < BrowserAction
  get "/users/:id" do
    # this gives me id still
  end
end
class User::Show < BrowserAction
  route do
     # this would give me user_id ?
   end
end
```

**Comment by paulcsmith** [comment]
Yeah that’s correct. Good clarification 👍

**Comment by paulcsmith** [comment]
Closed by #680 Thanks @bnjamin!
